app/MMB/src/components/sectionListScrollTo/SectionListScrollTo.js

```diff
@@ -0,0 +1,240 @@
+    } else {
+      this.layoutInfo[layoutInfoIndex].height = height;
+      this.layoutInfo[layoutInfoIndex].rendered = isRendered;
+    }
+  };
+
+  computeOffset = (index: number) => {
+    if (this.layoutInfo.every(item => item.rendered)) {
+      return this.layoutInfo
+        .slice(0, index)
+        .reduce((acc, curr) => acc + curr.height, 0);
+    }
+    return 0;
+  };
+
+  scrollToRelevantSection = () => {
+    const { data } = this.props;
+    const sectionToScroll = data.find(section => section.shouldScroll);
+    if (!data || !this.sectionListRef || !sectionToScroll) {
+      return;
+    }
+    const sectionIndex = data.indexOf(sectionToScroll);
+    const layoutInfoIndex = this.getLayoutInfoIndex(
+      0,
+      sectionToScroll,
+      'header',
+    );
+
+    const viewOffset = this.layoutInfo[layoutInfoIndex].height;
+    this.sectionListRef.scrollToLocation({
+      sectionIndex,
+      itemIndex: 0,
+      viewPosition: 0,
+      viewOffset,
+    });
+  };
+
+  verifyAndScroll = setInterval(() => {
+    if (this.layoutInfo.every(info => info.rendered)) {
+      const { data } = this.props;
+      const sectionToScroll = data.find(mysection => mysection.shouldScroll);
+      if (sectionToScroll) {
+        const layoutInfoIndex = this.getLayoutInfoIndex(
+          0,
+          sectionToScroll,
+          'header',
+        );
+        const currentOffset = this.computeOffset(layoutInfoIndex);
+
+        const layoutFinishedComputing = currentOffset === this.pastOffset;
+
+        if (layoutFinishedComputing) {
+          this.scrollToRelevantSection();
+          clearInterval(this.verifyAndScroll);
+        }
+        this.pastOffset = this.computeOffset(layoutInfoIndex);
+      }
+    }
+  }, 50);
+
+  getItemLayout = (data: ?(Section[]), index: number) => {
+    const ITEM_LENGTH_ESTIMATE = 100;
+    if (
+      this.layoutInfo[index] &&
+      this.layoutInfo.every(info => info.rendered)
+    ) {
+      const offset = this.computeOffset(index);
+      const length = this.layoutInfo[index].height;
+      return {
+        length,
+        offset,
+        index,
+      };
+    }
+    return {
+      length: ITEM_LENGTH_ESTIMATE,
+      offset: ITEM_LENGTH_ESTIMATE * index,
+      index,
+    };
+  };
+
+  renderSectionHeader = ({ section }: { section: Section }) => (
+    <OnLayout>
+      {({ height }) => {
+        this.addInfo('header', height, section);
+        return section.header;
+      }}
+    </OnLayout>
+  );
+
+  renderItem = ({ item, index, section }: RenderItem) => (
+    <OnLayout>
+      {({ height }) => {
+        this.addInfo('item', height, section, index);
+        return item;
+      }}
+    </OnLayout>
+  );
+
+  renderSectionFooter = ({ section }: { section: Section }) => {
+    this.addInfo('footer', 0, section);
+    return null;
+  };
+
+  keyExtractor = (item: React.Node[], index: number) => index;
+
+  componentDidMount() {
+    this.verifyAndScroll;
+  }
+
+  componentWillUnmount() {
```

Comment:
lifecycle methods should be the first methods of the class, in the order they are being called.